[10.0][FIX] web_widget_x2many_2d_matrix: Styles

houzefa-abba opened the pull request:
Hi,

I set out by wondering why cells in the matrix widget were no longer showing up with their blue "required field" background color: It appears various CSS classes (but not all of them) have been renamed in Odoo 10.

I then looked into other CSS classes used by the `web_widget_x2many_2d_matrix` addon and made them match regular Odoo lists.
In editable mode, we are still quite far from editable lists which put `input` fields inside a fake `div` so as to simulate an actual `form`, but styles do match now at least.

Don't hesitate to shout if anything is broken - I will be using this widget quite a bit so I will try to watch for issues reported in here.

Commit message:

Update CSS classes used by the "web_widget_x2many_2d_matrix" widget to
match base Odoo ones.

In particular, cells now show up with a "required field" style (as
intended).

Remove custom read-only field management in the process that was no
longer working and is not needed anymore (all handled via CSS classes).

houzefa-abba committed:
[10.0][FIX] web_widget_x2many_2d_matrix: Styles

Update CSS classes used by the "web_widget_x2many_2d_matrix" widget to
match base Odoo ones.

In particular, cells now show up with a "required field" style (as
intended).

Remove custom read-only field management in the process that was no
longer working and is not needed anymore (all handled via CSS classes).

pedrobaeza commented:
Can you include some screenshots of before/after?

houzefa-abba commented:
Good idea!

Here is what I get with the latest Odoo 10, `OCA/web`, `OCA/product-variant` after installing `sale_order_variant_mgmt`.

- Default look - notice the blue "required field" background is missing:
![variants-default](https://user-images.githubusercontent.com/6347423/33428712-3a1ad6be-d5ca-11e7-8451-a5513a0793f0.png)

- Default read-only look (I have added `readonly="1"` by hand) - notice all content is missing:
![variants-default-readonly](https://user-images.githubusercontent.com/6347423/33428745-628bf970-d5ca-11e7-9b36-188bce213cd2.png)

- With my fixes:
![variants-fixed](https://user-images.githubusercontent.com/6347423/33428760-6fb22a34-d5ca-11e7-83b1-c6351e864418.png)

- With my fixes in read-only mode:
![variants-fixed-readonly](https://user-images.githubusercontent.com/6347423/33428774-7d168134-d5ca-11e7-85f3-f8bc73ceda71.png)

These shots are with the default theme; I have been testing with [another theme](https://www.odoo.com/apps/themes/10.0/backend_theme_v10/) and it works fine there as well.

Also for each CSS class change, I `grep`ped through the `web` addon; usually the only result was in old docs that haven't been updated.

pedrobaeza commented on web_widget_x2many_2d_matrix/static/src/js/web_widget_x2many_2d_matrix.js:
```diff
@@ -207,13 +206,27 @@ odoo.define('web_widget_x2many_2d_matrix.widget', function (require) {
-            if(this.is_numeric)
-            {
-                classes += ' oe_form_field_float';
+            var classes = 'edit o_form_input o_form_required oe_edit_only';
```
But what if the field is not required?

houzefa-abba replied:
Would be easy to conditionally add the attribute based on `this.get('required')` but I'd rather not until the widget actually supports this; currently when I set the matrix to be required, it still allows being saved without any actual input.

This to me would be a separate improvement than these simple style fixes.

pedrobaeza replied:
OK

pedrobaeza commented on web_widget_x2many_2d_matrix/static/src/js/web_widget_x2many_2d_matrix.js:
```diff
@@ -380,17 +390,6 @@ odoo.define('web_widget_x2many_2d_matrix.widget', function (require) {
 
         },
 
-        effective_readonly_change: function()
```
Why this is not working?

houzefa-abba replied:
When testing this event was never fired; I think that is because the base `FieldX2Many` struct already catches it via:
```javascript
        this.is_loaded.done(function() {
            self.on("change:effective_readonly", self, destroy);
        });
```
(`web/.../form_relational_widgets.js`)

Since the whole widget is re-rendered on a `readonly` attr change, what this method was doing by hand is also no longer necessary.

pedrobaeza replied:
OK

pedrobaeza reviewed (changes requested):
Please change module version number according https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers

yajo commented:
Can't we make the style of 1 match the style of 2 with the default theme?

![33428760-6fb22a34-d5ca-11e7-83b1-c6351e864418](https://user-images.githubusercontent.com/973709/33477155-eae0bd0a-d684-11e7-8b80-5b23ba0d1376.png)

houzefa-abba commented:
Good catch @Yajo, that is because our `input` fields are missing a `type="text"` attribute which all other `input` fields have!

houzefa-abba committed:
[10.0][FIX] matrix widget: Add a missing attr

This "type" attribute is needed for default Odoo input styles to apply.

houzefa-abba committed:
[10.0][FIX] matrix widget: Disabled input fields

The background color of disabled input fields gets jumbled when
applying default Odoo styles. Reset it to grey (couldn't find a way to
reset it to the default "disabled" background color).

houzefa-abba commented:
Here is how it now looks.

I have also attempted to fix the disabled field bg color via CSS as doing it from the JS code would be annoying (the "disabled" attr gets added separately from CSS classes).

![variants-fixed-2](https://user-images.githubusercontent.com/6347423/33478355-4aa8e574-d689-11e7-9f7c-9ffee49490bc.png)

yajo reviewed (approved):
code looks good, no functional test done

pedrobaeza commented:
Please change module version number

ping @hbrunn

houzefa-abba committed:
[10.0] web_widget_x2many_2d_matrix: v 10.0.1.0.2

simahawk commented on web_widget_x2many_2d_matrix/static/src/xml/web_widget_x2many_2d_matrix.xml:
```diff
@@ -14,20 +14,22 @@
                 <tfoot t-if="widget.show_column_totals">
                     <tr>
                         <th>Total</th>
-                        <td t-foreach="widget.get_x_axis_values()" t-as="x" class="oe_list_footer oe_number column_total" t-att-data-x="x" />
```
`oe_list_footer` should be moved to `o_list_footer`. Is not used anywhere as far as I see but all the list-related classes are no `o_list_*`

simahawk reviewed (changes requested):
LGTM, one small non-blocking change in class name

ovnicraft commented:
any update for this ?
